lib/test.js

```diff
@@ -239,31 +233,45 @@ Test.prototype._publicApi = function () {
 		self._assert();
 	}
 
+	function onAssertionEvent(event) {
+		if (event.type === 'error') {
+			event.error.powerAssertContext = event.powerAssertContext;
+			event.error.originalMessage = event.originalMessage;
+			self._setAssertError(event.error);
+			self._assert();
+			return null;
+		}
+
+		var fn = observableToPromise(event.returnValue);
+
+		if (isPromise(fn)) {
+			return Promise.resolve(fn)
+				.catch(function (err) {
+					self._setAssertError(err);
```

Comment:
OK, then we are good. Those will never have `powerAssertContext`. Probably would be good to attach `originalMessage` though right?